src/Temporalio/Client/EnvConfig/ClientEnvConfig.cs

```diff
@@ -0,0 +1,414 @@
+    public sealed record ClientEnvConfig(IReadOnlyDictionary<string, ClientEnvConfig.Profile> Profiles)
+    {
+        /// <summary>
+        /// Load client configuration from environment variables and configuration files.
+        /// </summary>
+        /// <param name="options">Options for loading the configuration.</param>
+        /// <returns>Loaded configuration data.</returns>
+        public static ClientEnvConfig Load(ConfigLoadOptions options)
+        {
+            var runtime = Runtime.TemporalRuntime.Default.Runtime;
+            var profiles = Bridge.EnvConfig.LoadClientConfig(runtime, options);
+            return new ClientEnvConfig(profiles);
+        }
+
+        /// <summary>
+        /// Load client connection options directly from configuration.
+        /// </summary>
+        /// <param name="options">Options for loading the configuration profile.</param>
+        /// <returns>Client connection options.</returns>
+        public static TemporalClientConnectOptions LoadClientConnectOptions(ProfileLoadOptions options)
+        {
+            var clientProfile = Profile.Load(options);
+            return clientProfile.ToClientConnectionOptions();
+        }
+
+        /// <summary>
+        /// Convert to a dictionary structure that can be used for TOML serialization.
+        /// </summary>
+        /// <returns>Dictionary mapping profile names to their dictionary representations.</returns>
+        public IReadOnlyDictionary<string, Dictionary<string, object?>> ToDictionary() =>
+            Profiles.ToDictionary(
+                kvp => kvp.Key,
+                kvp => kvp.Value.ToDictionary());
+
+        /// <summary>
+        /// Create a ClientEnvConfig from a dictionary structure.
+        /// </summary>
+        /// <param name="profileDictionaries">Dictionary of profile name to profile dictionary.</param>
+        /// <returns>Client configuration instance.</returns>
+        public static ClientEnvConfig FromDictionary(IReadOnlyDictionary<string, Dictionary<string, object?>> profileDictionaries)
+        {
+            var profiles = profileDictionaries.ToDictionary(
+                kvp => kvp.Key,
+                kvp => Profile.FromDictionary(kvp.Value));
+
+            return new ClientEnvConfig(profiles);
+        }
+
+        /// <summary>
+        /// Represents a client configuration profile.
+        /// </summary>
+        /// <param name="Address">Client address.</param>
+        /// <param name="Namespace">Client namespace.</param>
+        /// <param name="ApiKey">Client API key.</param>
+        /// <param name="Tls">TLS configuration.</param>
+        /// <param name="GrpcMeta">gRPC metadata.</param>
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Naming", "CA1724:Type names should not match namespaces", Justification = "Profile is appropriate name in this context and unlikely to cause confusion")]
+        public sealed record Profile(
```

Comment:
```suggestion
#pragma warning disable CA1724 // We're ok with the Profile name since it's a nested class
        public sealed record Profile(
#pragma warning restore CA1724
```

I think this is the better/clearer way